### DISCUSSION TITLE
httpx 0.24 documentation issue for netrc support

### DISCUSSION THREAD

**Discussion by alexprengere** [encode/httpx#2665] [Potential Issue] [answered]
From the docs: https://www.python-httpx.org/advanced/#netrc-support
```
>>> auth = httpx.NetRCAuth()
>>> client = httpx.Client(auth=auth)
```
Using that snippet will yield the error:
```
>>> httpx.NetRCAuth()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: NetRCAuth.__init__() missing 1 required positional argument: 'file'
```
Using `NetRCAuth(file=None)` fixes the issue, and produces the expected behavior, but perhaps it should be made the default value (I can open a PR).

**Comment by lovelydinosaur** [comment]
> Using NetRCAuth(file=None) fixes the issue, and produces the expected behaviour, but perhaps it should be made the default value.

Oops, yes it's typed as `Optional`, and I'd agree `None` *should* be the default, in line with the docs.

> I can open a PR.

If you'd like to do that please do go ahead yup.

**Reply by lovelydinosaur** [reply, marked as answer]
Escalated to #2666

**Reply by alexprengere** [reply]
PR opened, thanks for your reactivity!
